src/v/iceberg/values_bytes.cc

```diff
@@ -0,0 +1,157 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#include "iceberg/values_bytes.h"
+
+#include "bytes/iobuf_parser.h"
+
+#include <variant>
+
+namespace iceberg {
+
+namespace {
+
+struct value_to_bytes_visitor {
+    template<typename T>
+    requires(std::is_scalar_v<std::decay_t<decltype(T::val)>>)
+    bytes operator()(const T& v) {
+        iobuf ret;
+        if constexpr (sizeof(T::val) == 1) {
+            ret.append(reinterpret_cast<const char*>(&v.val), 1);
+        } else if constexpr (std::is_same_v<decltype(T::val), float>) {
+            const auto le_t = htole32(std::bit_cast<std::uint32_t>(v.val));
+            ret.append(reinterpret_cast<const char*>(&le_t), sizeof(le_t));
+        } else if constexpr (std::is_same_v<decltype(T::val), double>) {
+            const auto le_t = htole64(std::bit_cast<std::uint64_t>(v.val));
+            ret.append(reinterpret_cast<const char*>(&le_t), sizeof(le_t));
+        } else {
+            const auto le_t = ss::cpu_to_le(v.val);
+            static_assert(sizeof(le_t) == sizeof(T::val));
+            ret.append(reinterpret_cast<const char*>(&le_t), sizeof(le_t));
```

Comment:
NOTE: almost identical to the code we have in serde/rw/scalar.h

Reply:
if it is actually different then i think it's worth having separate implementations to keep things simple to understand.